packages/visual-editor/src/components/EditorBlock/EditorBlock.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from 'react';
+import { ASSEMBLY_NODE_TYPE } from '@contentful/experiences-core/constants';
+import {
+  ComponentRegistration,
+  ExperienceTreeNode,
+  ResolveDesignValueType,
+} from '@contentful/experiences-core/types';
+import {
+  CircularDependencyErrorPlaceholder,
+  MissingComponentPlaceholder,
+  ImportedComponentErrorBoundary,
+} from '@/error-handling';
+import styles from './EditorBlock.module.css';
+import { useComponentRegistration } from './useComponentRegistration';
+import { useComponentProps } from './useComponentProps';
+
+type NodeBlockProps = {
+  node: ExperienceTreeNode;
+  resolveDesignValue: ResolveDesignValueType;
+  wrappingPatternIds?: Set<string>;
+};
+
+export function EditorBlock({
+  node,
+  resolveDesignValue,
+  wrappingPatternIds: parentWrappingPatternIds = new Set(),
```

Comment:
on each rerender this default prop will be a `new Set()`, which will trigger recalculation of the `useMemo`, but maybe it's fine in this case